### ISSUE TITLE
MQTT5 CONNECT and PUBLISH Properties

### ISSUE THREAD

**Issue by pearmaster** [asyncapi/bindings#199]
#### Reason/Context
One of the defining features of MQTTv5 are _Properties_.  (otherwise, you'd probably just use MQTT v3).  Properties can be included in almost all MQTT packet types, and can form a significant part of an MQTTv5 API contract.  

In issue https://github.com/asyncapi/spec/issues/878 @fmvilas suggested that `bindings` are the appropriate construct for describing _properties_. 

For more information, this page lists properties by packet type: http://www.steves-internet-guide.com/mqttv5-properties/

#### Description
I suggest adding additional fields to the MQTT5 server binding for MQTT CONNECT packets:
 * Maximum Packet Size
 * User Property
 * Session Expiry Interval (already included in the MQTT5 0.2.0 binding)

I suggest adding additional fields to the MQTT5 message bindings for MQTT PUBLISH packets:
 * Paylod Format Indicator
 * Message Expiry Interval
 * Response Topic
 * Correlation Data
 * ~User Property~ (These are OK to be message headers per https://github.com/asyncapi/spec/issues/878#issuecomment-1328564400)
 * Content Type

I have a pull request that is about ready to go once this issue is approved (according to the contributing guidelines)

_Side note: I've used the word "packet" above when the correct MQTT terminology is "message" to disambiguate from AsyncAPI "message" objects._

**Comment by pearmaster** [comment]
@fmvilas While I'm looking at this, I see that `mqtt5` doesn't include the bindings that `mqtt` allows.  Do you envision bindings being able to stack like this?:
```yaml
servers:
  prod:
    bindings:
      mqtt:
        clientId: guest
      mqtt5:
        sessionExpiryInterval: 60
```

or should the relevant `mqtt` bindings be copied to the set of `mqtt5` bindings?

**Comment by fmvilas** [comment]
> Do you envision bindings being able to stack like this?

I think we somehow defined `mqtt` for MQTT <5 and `mqtt5` for version 5 but, to be honest, I think this is a mistake we did in the past and both bindings should be merged into `mqtt`. It should be ok to put MQTT 5 stuff into the `mqtt` binding. We should just make it clear that it's only supported in versions >=5.

**Comment by pearmaster** [comment]
I'll adapt my MR to merge both binding definitions (plus the new stuff) into `mqtt`.

**Comment by fmvilas** [comment]
Awesome, thanks! 🙏
